libiocage/lib/RCConf.py

```diff
@@ -29,38 +29,71 @@
 
 
 class RCConf(dict):
-    def __init__(self, path, data={}, logger=None, jail=None):
+
+    # the file is always relative to the resource
+    _file: str = "/etc/rc.conf"
+
+    def __init__(
+        self,
+        resource: 'libiocage.lib.Resource.LaunchableResource',
+        file: str=None,
+        logger: 'libiocage.lib.Logger.Logger'=None
+    ) -> None:
 
         dict.__init__(self, {})
         self.logger = libiocage.lib.helpers.init_logger(self, logger)
-        self.jail = jail
 
         # No file was loaded yet, so we can't know the delta yet
         self._file_content_changed = True
-        self._path = None
-        self.path = path
+
+        if file is not None:
+            self._file = file
+
+        self.resource = resource
+        self._read_file()
 
     @property
     def path(self):
-        return object.__getattribute__(self, "_path")
+        path = f"{self.resource.root_dataset.mountpoint}/{self.file}"
+        return os.path.abspath(path)
```

Comment:
does this actually "validate" the path, or does it just normalize it?

Reply:
💥  You've spotted a vulnerability! We should never trust files a jail's directory. The file could be a symlink pointing to arbitrary locations for example. Thanks for bringing it up!

Reply:
are symlinks even able to leave a jail boundary?

Reply:
We are writing to this file from the host with root permission. So if the rc.conf is a symlink to an absolute path, our host would write the changes to the link target, instead of the location within the jail's root dataset. Not good!